lib/guard/api.rb

```diff
@@ -0,0 +1,279 @@
+require "guard"
+require "guard/internals/groups"
+
+module Guard
+  module API
+    TEMPLATE_FORMAT = "%s/lib/guard/%s/templates/Guardfile"
+
+    require "guard/ui"
+
+    def self.included(base)
+      base.extend(ClassMethods)
+      base.class_eval do
+        attr_accessor :group, :watchers, :callbacks, :options
+      end
+    end
+
+    module ClassMethods
+      # Get all callbacks registered for all Guard plugins present in the
+      # Guardfile.
+      #
+      def callbacks
+        @callbacks ||= Hash.new { |hash, key| hash[key] = [] }
```

Comment:
I think Ruby will complain about this as it wasn't defined prior. Unfortunately, I think you need to do something like this: https://github.com/ioquatix/utopia/blob/9239e84edfeaf7177f74ae9408c4e16468faa505/lib/utopia/controller/actions.rb#L126-L131

Check by running tests with warnings turned on.